File: Backend/src/main/java/org/example/datn_sd69/modules/category/service/CategoryService.java
package org.example.datn_sd69.modules.category.service;

import org.example.datn_sd69.entity.Category;
import org.example.datn_sd69.modules.category.dto.request.CategoryRequest;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface CategoryService {
    List<Category> getAll();
    Category getById(Integer id);
    Category create(CategoryRequest request);
    Category update(Integer id, CategoryRequest request);
    void delete(Integer id);
    List<Category> search(String keyword);
    Page<Category> getAll(Pageable pageable);
    Page<Category> search(String keyword, Pageable pageable);

    // --- THÊM DÒNG NÀY: Lấy danh sách cho khách xem (chỉ hàng đang hoạt động) ---
    Page<Category> getActiveCategories(String keyword, int page, int size);
}

File: Frontend/src/modules/admin/feature/category/services/category.service.ts
import api from '@/common/api'; 
import type { Category, CategoryRequest } from '../types/category.type';

export const categoryService = {
  getAll(config?: any) {
    return api.get('/admin/categories', config);
  },
  
  getById(id: number) {
    return api.get(`/admin/categories/${id}`);
  },

  create(data: CategoryRequest) {
    return api.post('/admin/categories', data);
  },

  update(id: number, data: CategoryRequest) {
    return api.put(`/admin/categories/${id}`, data);
  },

  delete(id: number) {
    return api.delete(`/admin/categories/${id}`);
  }
};